Scripts/Game/ODIN/Editor/Containers/Actions/ContextActions/ODIN_TeleportToSquadmateContextAction.c

```diff
@@ -0,0 +1,134 @@
+	{
+		if (!hoveredEntity)
+			return;
+		
+		GenericEntity owner = hoveredEntity.GetOwner();
+		
+		MenuBase myMenu = GetGame().GetMenuManager().OpenDialog(ChimeraMenuPreset.ODIN_ListBox_ID, DialogPriority.INFORMATIVE, 0, true); 
+		
+		// null check 
+		if (!myMenu)
+		{
+			ODIN_LogHelper.Log("Could not find listbox dialog defined", "TeleportToSquadmateContextAction", LogLevel.WARNING);
+			return;	
+		}
+		
+		// cast so we have dialog object. Set hovered-entity, we need to get squadmates from it and have it to teleport it
+		ODIN_ListboxSingleSelectDialog myMenuUI = ODIN_ListboxSingleSelectDialog.Cast(myMenu);
+		myMenuUI.SetHoveredEntity(hoveredEntity);
+		
+		// set Title 
+		myMenuUI.SetTitle("#ODIN-ActionList_TeleportToSquadmate_Name");
+			
+		// Get squadmates 
+		SCR_AIGroup group = ODIN_GroupHelper.GetGroup(owner);
+		if (!group || group.GetPlayerCount() <= 0) 
+		{
+			// TODO give error message back to zeus that unit is not in a group 
+			ODIN_LogHelper.Log("null group or <= 0 in group", "TeleportToSquadmateContextAction", LogLevel.WARNING);
+			return;
+		}
+		
+		// can't get character entities directly from group, only for AI. So for players we get their IDs and gotta find and cast them with that
+		array<int> members = group.GetPlayerIDs();
+		foreach (int playerID : members)
+		{
+			PlayerManager playerManager = GetGame().GetPlayerManager();
+			string name = playerManager.GetPlayerName(playerID);
+			ChimeraCharacter player = ChimeraCharacter.Cast(playerManager.GetPlayerControlledEntity(playerID));
+			
+			// if player is dead and the object is null, we skip
+			if (!player)
+				continue;
+			
+			// don't add the unit we want to teleport
+			if (player.GetID() == owner.GetID())
+				continue;
+			
+			// if Leader, show it
+			if (group.IsPlayerLeader(playerID))
+				name += " (Leader)";
+			
+			// if in Vehicle, show it
+			if (player.IsInVehicle())
+				name += " [VEHICLE]";
+			
+			myMenuUI.AddItemToListbox(name, player);
+		}
+		
+		// add action to confirm 
+		myMenuUI.GetOnConfirm().Insert(ODIN_TeleportToSquadmateContextAction.ODIN_OnConfirm);
+	}
+	
+	// function for script invoker
+	static void ODIN_OnConfirm(SCR_EditableEntityComponent hoveredEntity, SCR_ListBoxComponent listbox)
+	{
+		// todo, get selected player 
+		int selectedItem = listbox.GetSelectedItem();
+		if (selectedItem == -1)
+		{
+			ODIN_LogHelper.Log("Listbox returned -1 as selection, seemed like no option was selected", "TeleportToSquadmateContextAction", LogLevel.WARNING);
+			return;
+		}
+		
+		ChimeraCharacter selectedPlayer = ChimeraCharacter.Cast(listbox.GetItemData(selectedItem));
+		if (!selectedPlayer)
+		{
+			ODIN_LogHelper.Log("ChimeraCharacter in userdata for selected listbox item is null", "TeleportToSquadmateContextAction", LogLevel.WARNING);
+			return;
+		}
+
+		//if (selectedPlayer.IsInVehicle())
+		//{
+			// try to put hoveredentity inside vehicle? otherwise keep going and put it next to vic 
```

Comment:
Check out `SCR_PlayerSpawnPoint`, which is for making players spawn points. It has the handling for players in vehicles as well.

Reply:
Thanks for the hint! That did the trick.